sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -126,17 +126,15 @@ static inline _nib_onl_entry_t *_cache_out_onl_entry(const ipv6_addr_t *addr,
             /* cstate masked in _nib_nc_add() already */
             res->info |= cstate;
             res->mode = _NC;
-            clist_rpush(&_next_removable, (clist_node_t *)res);
-            break;
         }
-        /* requeue if not garbage collectible at the moment */
-        DEBUG("nib: Requeing (addr = %s, iface = %u)\n",
-              ipv6_addr_to_str(addr_str, &tmp->ipv6,
-                               sizeof(addr_str)),
-              _nib_onl_get_if(tmp));
+        /* requeue if not garbage collectible at the moment or queueing
+         * newly created NCE */
         clist_rpush(&_next_removable, (clist_node_t *)tmp);
-        tmp = (_nib_onl_entry_t *)clist_lpop(&_next_removable);
-    } while (tmp != first);
+        if (res == NULL) {
+            /* no new entry created yet, get next entry in FIFO */
+            tmp = (_nib_onl_entry_t *)clist_lpop(&_next_removable);
+        }
+    } while ((tmp != first) && (res != NULL));
```

Comment:
`.text` sizes:

board (arch) | without 0bcd911 | with 0bcd911 | diff
------------ | --------------- | ------------ | -------
arduino-uno (AVR)|19676|19628|48
iotlab-m3 (ARM Cortex-M3)|22264|22236|28
pic32-clicker (MIPS)|90580|90536|44
samr21-xpro (ARM Cortex-M0)|22556|22536|20
z1 (MSP430)|21086|21058|28
native (x86)|50137|50153|-16

Just for notice, the versions strings are

* **without 0bcd911**: `2017.04-devel-1280-gb2685-mk-gnrc_ipv6_nib/fix/nc-cache-out` (reverted 0bcd911, that's why its not bdd421c)
* **with 0bcd911**: `2017.04-devel-1279-g0bcd91-mk-gnrc_ipv6_nib/fix/nc-cache-out`

so the size benefit doesn't come from them.

I'd argue however that the size benefit is large enough to keep 0bcd911. What do you say @BytesGalore, @cgundogan.